esp/platform/espcontext.cpp

```diff
@@ -87,6 +87,19 @@ class CEspContext : public CInterface, implements IEspContext
 
     StringAttr   m_transactionID;
 
+    void flushTraceSummary()
```

Comment:
This method was public, and only manipulated the in-memory content without updating the log file for a specific reason. At some point, for legacy compatibility, we may need to generate the current state of the TxSummary prior to destruction of the context. The specific use case is an ESP request option that causes the TxSummary to be added to the response output. In this case we would want the data to be as correct as possible, with the knowledge that some values might change slightly between putting the data in the response and destroying the context.

What did this refactoring get us?

Reply:
Based on our discussion today, I will leave the flushTraceSummary() to be public so that it can be called outside. I will move the log() to the ~CEspContext().

I will remove the check 'if (getTxSummaryLevel() >= LogMin)' and move the LogMin to both the set and the append. Also do the similar inside the updateTraceSummaryHeader().